### ISSUE TITLE
Alerting: No data alerts fires two instance for one rule for classic condition, due to labelling issue.

### ISSUE THREAD

**Issue by sirdidymus** [grafana/grafana#71993]
### What happened?

Classic condition rule with expression set as alert criteria, sets labels for dashboard_uid and ref_id ,  which instantiate multiple instance for same rule. 

When we hit NO DATA state , (nodata=nodata),  we receive notification that two instance have hit nodata state, instead of the one (for the one rule) 

issue appears to be caused by this labelling issue https://github.com/grafana/grafana/issues/71467 

discussed https://raintank-corp.slack.com/archives/C01LJ5F8NRX/p1689722105737669  and raising issue as requested. 

Classic condition:
![image](https://github.com/grafana/grafana/assets/30308766/0ffdf031-2a2d-47cc-b7bc-eaeb152ce179)
two instances with unique ref_id and datasource_uid:
![image](https://github.com/grafana/grafana/assets/30308766/9d87b631-38d1-446b-8c77-a2ca4f036c66)
multiple notification:
![image](https://github.com/grafana/grafana/assets/30308766/78588758-30be-429d-ab03-12693373027d)

### What did you expect to happen?

When hitting nodata state for single rule, we should only get notified of one instance of state. 

### Did this work before?

-

### How do we reproduce it?

1.write classic condition, set alert criteria to an expression; set nodata=nodata or error=error
2.hit nodata or error state ,  observe rule dimensions and unique labelling
3.check notifications and see duplicates. 

### Environment (with versions)?

Grafana:
OS:
Browser:

### Grafana platform?

I use Grafana Cloud

**Comment by jalaziz** [comment]
I don't believe this is limited to classic conditions. We're seeing the same issue but for the Loki datasource and regular expressions (reduce and threshold).
